cpu/atmega_common/thread_arch.c

```diff
@@ -245,7 +244,7 @@ void NORETURN __enter_thread_mode(void)
     UNREACHABLE();
 }
 
-void thread_arch_yield(void) {
+void thread_yield_higher(void) {
```

Comment:
was thread_arch_yield the same as thread_yield_higher? The name suggest that it should be thread_yield instead of _higher.

Reply:
I agree the naming is unfortunate.
I checked, ```thread_arch_yield()``` is the same as ```thread_yield_higher()```. For all platforms, it saves the context, runs ```sched_run()``` and then restores ```sched_active_thread()```

```thread_yield()``` is defined in ```thread.c```, and advances the circular list of the current priority, then calls ```thread_yield_higher()```.